### PR TITLE
pkg: delete config when QPS and concurrency are both deleted

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #4373.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/ratelimit/option.go [hunk 1]
```diff
@@ -78,7 +78,11 @@ func UpdateDimensionConfig(cfg *DimensionConfig) Option {
 			return InAllowList
 		}
 		lim, _ := l.limiters.LoadOrStore(label, newLimiter())
-		return lim.(*limiter).updateDimensionConfig(cfg)
+		status := lim.(*limiter).updateDimensionConfig(cfg)
+		if status&QPSDeleted != 0 && status&ConcurrencyDeleted != 0 {
+			l.limiters.Delete(label)
```
Comment (lhy1024) [inline pkg/ratelimit/option.go:83]: What will happen if this is not removed?

Comment (rleungx) [reply pkg/ratelimit/option.go:83]: leave a config in output that is meaningless

**Review by lhy1024 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Comment by rleungx** [comment]
I made more changes. PTAL @lhy1024 @okJiang

File: pkg/ratelimit/option.go [hunk 2]
```diff
@@ -89,6 +83,6 @@ func InitLimiter() Option {
 			return InAllowList
 		}
 		l.limiters.LoadOrStore(label, newLimiter())
-		return ConcurrencyChanged
+		return LimiterNotChanged
```
Comment (okJiang) [inline pkg/ratelimit/option.go:86]: Has there been a change here, as expected?

File: pkg/ratelimit/limiter.go [hunk 3]
```diff
@@ -89,47 +82,51 @@ func (l *limiter) getConcurrencyLimiterStatus() (limit uint64, current uint64) {
...
-		return QPSDeleted
+		return LimiterNotChanged
 	}
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	if l.rate != nil {
+		if limit <= eps || burst < 1 {
+			l.rate = nil
+			if l.isEmpty() {
+				return LimiterDeleted
```
Comment (okJiang) [inline pkg/ratelimit/limiter.go:116]: Should this be `LimiterNoChanged`?

File: pkg/ratelimit/option.go (hunk 2 above)
Comment (rleungx) [reply pkg/ratelimit/option.go:86]: Yes, the limit has not been changed when we initialize the limiter.

File: pkg/ratelimit/limiter.go (hunk 3 above)
Comment (rleungx) [reply pkg/ratelimit/limiter.go:116]: No, LimiterDeleted indicates that we should remove the corresponding label from the config.

**Review by lhy1024 — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**